src/components/MentionSuggestions.tsx

```diff
@@ -1,73 +1,71 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import {View} from 'react-native';
-import _ from 'underscore';
 import getStyledTextArray from '@libs/GetStyledTextArray';
 import styles from '@styles/styles';
 import * as StyleUtils from '@styles/StyleUtils';
 import themeColors from '@styles/themes/default';
 import CONST from '@src/CONST';
 import AutoCompleteSuggestions from './AutoCompleteSuggestions';
 import Avatar from './Avatar';
-import avatarPropTypes from './avatarPropTypes';
 import Text from './Text';
 
-const propTypes = {
-    /** The index of the highlighted mention */
-    highlightedMentionIndex: PropTypes.number,
+// TODO: remove when avatarPropTypes will be migrated to TS
+type AvatarFunction = () => void;
 
-    /** Array of suggested mentions */
-    mentions: PropTypes.arrayOf(
-        PropTypes.shape({
-            /** Display name of the user */
-            text: PropTypes.string,
+type AvatarIcon = {
```

Comment:
in OnyxCommon module we have already Icon component which have same props , so you can use it instead of creating this type

Reply:
Not sure that I can use it 

These types are different in fields and types

Reply:
I've defined the type with the same signature that was used before (avatarPropTypes.js)

Reply:
I am almost certain that every Icon which we use in the app should be of type which you created, so I would adjust type of the Icon in OnyxCommon

Reply:
As I see we do not use this type from the OnyxCommon anywhere and it looks like we will have this type under types/onyx/avatar or smth like that

Reply:
@blazejkustra what do you think? 

I see that you authored this commit with Icon type creation in the OnyxCommon

Reply:
> As I see we do not use this type from the OnyxCommon anywhere and it looks like we will have this type under types/onyx/avatar or smth like that

It's used in Report.ts

After digging in the code I found where suggestions are created - [here](https://github.com/pasyukevich/expensify-app/blob/0138d943fc97c4576e41cb0fd937bb8b7d7bf2d0/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L141) and [here](https://github.com/pasyukevich/expensify-app/blob/0138d943fc97c4576e41cb0fd937bb8b7d7bf2d0/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L166).

I think you should be able to adjust the type in OnyxCommon like Kuba suggested